test/test_events.py

```diff
@@ -0,0 +1,87 @@
+# Copyright 2025 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from __future__ import annotations
+
+import typing
+
+import pytest
+
+import ops
+from ops import Framework
+from ops.testing import CharmEvents, Context, State
+
+
+def collect_filtered(
+    *event_type: type[ops.EventBase], meta: dict[str, typing.Any] | None = None
+) -> tuple[ops.EventBase, ...]:
+    ctx = Context(ops.CharmBase, meta={'name': 'ben', **(meta or {})})
+    with ctx(ctx.on.update_status(), State()) as mgr:
+        return tuple(e.event_type for e in mgr.charm.on.events(*event_type).values())  # type: ignore
+
+
+def test_filtering_null():
```

Comment:
I find this test hard to follow. What's it actually testing, and can we rewrite it in a simpler, more imperative way?